src/android/CardIoPlugin.java

```diff
@@ -45,7 +49,26 @@ public boolean execute(String action, JSONArray args, CallbackContext callbackCo
      * @param args
      * @throws JSONException
      */
-    private boolean scanCard(CallbackContext callbackContext, JSONArray args) throws JSONException {
+    private boolean scanCard(CallbackContext callbackContext, final JSONArray args) throws JSONException {
+
+        if (!CardIOActivity.canReadCardWithCamera()) {
+            this.cordova.requestPermission(new CordovaPlugin(){
+                @Override
+                public void onRequestPermissionResult(int requestCode, String[] permissions, int[] grantResults) throws JSONException {
+                    super.onRequestPermissionResult(requestCode, permissions, grantResults);
+
+                    if(grantResults[0] == PackageManager.PERMISSION_GRANTED)
```

Comment:
What happens if the grantResults array is null?

Reply:
..exceptions! Line fixed with a new commit, thanks!